Hide hidden files

BoraxTheClean opened the pull request:
- Hide Hidden Files
- Display a prettier directory name rather than `.thought_box`

BoraxTheClean committed:
Don't display hidden files

BoraxTheClean committed:
Display  instead of  when browsing the root of your notes directory

BoraxTheClean committed:
Prevent file names from starting with a

standard-affect commented on src/custom_types/scroll_menu.py:
```diff
@@ -97,11 +101,14 @@ def _get_contents(self, directory: str) -> List[Frame]:
             )
-            for item in listdir(directory)
+            for file_name in listdir(directory)
+            if not file_name.startswith(".")
```
If I'm reading right the user would still be able to create a file that starts with `.` as long as they enter a path starting from a directory inside `.thought_box` that doesn't itself start with `.` Should we prevent this?

BoraxTheClean replied:
I believe this was addressed in 6833501c3cde74adeff3502ce271317870e08e70.

BoraxTheClean replied:
If you're seeing another way around it, please elaborate. Or record a loom of you doing the thing.

SharpBit replied:
> If I'm reading right the user would still be able to create a file that starts with `.` as long as they enter a path starting from a directory inside `.thought_box` that doesn't itself start with `.` Should we prevent this?

This should be fixed now in f8b34c4756b1f341bf453543a6a694bc87b791de

standard-affect reviewed (commented):
Good work, though per my comment it might be possible for users to evade the dotfile restriction.

SharpBit committed:
only show .txt and .md files as options to open

- automatically make file .txt if not specified or is some other ext.
- fix bug where current path was being updated even if path was invalid

SharpBit commented:
> Good work, though per my comment it might be possible for users to evade the dotfile restriction.

![image](https://user-images.githubusercontent.com/31069084/125837599-97a31da5-8380-4e59-8c86-59b0978a9f77.png)
It seems to be possible, yes.

SharpBit committed:
fix being able to make a file that starts with "." inside another folder

SharpBit committed:
Merge branch 'main' into hide-hidden-files

standard-affect reviewed (approved):
Only setting `application_state.current_path` if the save is successful is a good idea.

SharpBit committed:
add back folders to the open scroll menu

SharpBit committed:
Merge branch 'hide-hidden-files' of https://github.com/CupOfGeo/adaptable-antelopes into hide-hidden-files
